### PR TITLE
Make Powershell stop on errors [depends-on: #3616, blocks: #2310, #3636, #3637]

### PR DESCRIPTION
**Pull request by tautschnig**
We had failing regression tests, but this went unnoticed because the goto-cl
regression tests were successful (which were the last common run in the cmdlet).

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 10aeffead84ccaaadc4d24dcb343025792c099e5).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/95900880

**Review by kroening — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: aeded5401f28f91cf34b8d6dcbf737f9c0c56428).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/95911410

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 430c6a6e264141a320e04fdc14996a52229850e2).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/95993085

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 2d9c00b504e4e35c08ef496b201aa605fd3e39e9).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/96062013

File: buildspec-windows.yml [hunk 1]
```diff
@@ -64,6 +64,9 @@ phases:
       - |
         $env:Path = "C:\tools\cygwin\bin;$env:Path"
         cmd /c 'call "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat" x64 && bash -c "make -C regression test BUILD_ENV=MSVC" '
+
+      - |
```
Comment (chrisr-diffblue) [inline buildspec-windows.yml:68]: Apologies if I don't understand the buildspec + windows combination - but why does `regression/goto-cl` need to be treated any differently from any of the other subdirectories in `regression/` ?

Comment (smowton) [reply buildspec-windows.yml:68]: I think it's just that `goto-cl` is not part of the standard `make test` target set. Perhaps we should make it so under win32, either by checking $OS or by checking if `cl.exe` appears to be in the path using `which`.

Comment (chrisr-diffblue) [reply buildspec-windows.yml:68]: Fixing up the `make test` target to understand that on win32 or whatever, extra targets should be added would probably be preferable/less error prone when moving between CI infrastructure, etc.

Comment (tautschnig) [reply buildspec-windows.yml:68]: I'll take care of this in a separate PR. We also have some weird interaction between scripts removing dirs and the makefiles at the moment, which should all go away thanks to the -X/-I options in test.pl.

**Pull request merged**